src/objects/zcl_abapgit_object_oa2p.clas.abap

```diff
@@ -0,0 +1,258 @@
+                  CHANGING cg_data = ls_profile ).
+
+    io_xml->read( EXPORTING iv_name = 'SCOPES'
+                  CHANGING cg_data = lt_scope ).
+
+
+    MODIFY oa2c_profiles FROM ls_profile.
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( |Error updating table OA2C_PROFILES when deserialize profile { mv_profile }.| ).
+    ENDIF.
+
+    MODIFY oa2p_scopes FROM TABLE lt_scope.
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( |Error updating table OA2P_SCOPES when deserialize profile { mv_profile }.| ).
+    ENDIF.
+
+    tadir_insert( iv_package ).
+
+    corr_insert( iv_package ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~exists.
+
+    SELECT SINGLE profile FROM oa2c_profiles
+      INTO mv_profile
+      WHERE profile = mv_profile.                       "#EC CI_GENBUFF
+    rv_bool = boolc( sy-subrc = 0 ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_comparator.
+    RETURN.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_deserialize_steps.
+    APPEND zif_abapgit_object=>gc_step_id-abap TO rt_steps.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_metadata.
+    rs_metadata = get_metadata( ).
+    rs_metadata-delete_tadir = abap_true.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_active.
+    rv_active = is_active( ).
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_locked.
+
+    DATA: lv_profile_name TYPE eqegraarg,
+          lv_lock_number  TYPE int4,
+          lt_locks        TYPE  STANDARD TABLE OF seqg3.
+
+    lv_profile_name = mv_profile.
+
+    CALL FUNCTION 'ENQUEUE_READ'
+      EXPORTING
+        gclient = sy-mandt    " Client
+        gname   = 'OA2C_PROFILES'    " Granularity name (-> table name)
+        garg    = lv_profile_name    " Granularity value(->values of key fields)
+      IMPORTING
+        number  = lv_lock_number
+      TABLES
+        enq     = lt_locks.    " Number of chosen lock entries
+
+
+    IF lv_lock_number > 0.
+      rv_is_locked = abap_true.
+    ELSE.
+      rv_is_locked = abap_false.
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~jump.
+    zcx_abapgit_exception=>raise( |Jump to infoObjects is not yet supported| ).
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~serialize.
+
+    DATA: ls_profile TYPE oa2c_profiles,
+          lt_scope   TYPE STANDARD TABLE OF oa2p_scopes.
+
+
+    SELECT SINGLE * FROM oa2c_profiles
+                    INTO ls_profile
+                    WHERE profile = mv_profile.         "#EC CI_GENBUFF
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise(
+        |Error when geting getails of OAuth2 Profile { mv_profile } from table OA2C_PROFILES.| ).
+    ENDIF.
+
+    SELECT * FROM oa2p_scopes
+             INTO TABLE lt_scope
+             WHERE profile = mv_profile.                "#EC CI_GENBUFF
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise(
+        |Error when geting Scopes of OAuth2 Profile { mv_profile } from table OA2C_SCOPES.| ).
+    ENDIF.
+
+    io_xml->add( iv_name = 'PROFILE'
```

Comment:
created_ by/on/at and changed_ by/on/at should be cleared here to prevent diffs.

https://github.com/abapGit-tests/OA2P/blob/1c134142e89dc6dd32e66c8dc9f4e795a66cc58e/src/ztest_profile.oa2p.xml#L8-L13

Reply:
I will fix it.